extension/persistence/eclipselink/src/main/java/org/apache/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkMetaStoreSessionImpl.java

```diff
@@ -257,6 +259,12 @@ public <T> T runInTransaction(
       } finally {
         localSession.remove();
       }
+    } catch (PersistenceException e) {
+      if (e.toString().toLowerCase(Locale.ROOT).contains("duplicate key")) {
+        throw new IllegalStateException("Duplicate key error when persisting entity", e);
```

Comment:
`PersistenceException` is not visible in the service, and Iceberg's `AlreadyExistsException` is not visible here. It might be ideal to introduce our own exception mapper with our own exceptions kept in polaris-core, but that's a big migration. 

For now, I'm using this `IllegalStateException` to negotiate between the two exception types.

Reply:
It looks like we do already have some specializations for specific jakarta exceptions in [IcebergExceptionMapper](https://github.com/apache/polaris/blob/8f14363abc42ea6d95b67bc407f825b0af9bb4f6/polaris-service/src/main/java/org/apache/polaris/service/IcebergExceptionMapper.java#L71) such as for:

    case jakarta.ws.rs.ForbiddenException e -> Response.Status.FORBIDDEN.getStatusCode();

There's also a kind of catch-all for:

    case WebApplicationException e -> e.getResponse().getStatus();

To avoid throwing a 500 `INTERNAL_SERVER_ERROR` maybe we can either throw `jakarta.persistence.EntityExistsException` here and add a case there to translate that into `CONFLICT` or we could throw a generic `WebApplicationException` here with `CONFLICT` status code already set.

Reply:
I like the idea of using `jakarta.persistence.EntityExistsException` but it seems like that dependency might not be there yet. I will try the latter approach.

<img width="400" alt="Screenshot 2024-09-06 at 12 38 38 AM" src="https://github.com/user-attachments/assets/999b1ea1-f3ea-4cca-8a93-25a2c70ee1d0">

Reply:
It seems that `jakarta.ws.rs.WebApplicationException` is also not available right now. I have tightened the catch to reduce the likelihood that a different `IllegalStateException` gets caught here, but I agree that ideally we could just throw a mapped exception here. Perhaps we can add a new exception mapper?

Reply:
Do we still need to catch it just to rethrow in `PolarisServiceImpl`? Seems we should just let it bubble all the way out adn the `IcebergExceptionMapper` will handle it automatically. Or does `IcebergExceptionMapper` not kick in for these for some reason?

Reply:
If the problem is that `IcebergExceptionMapper` can't see `jakarta.persistence`, I think it's reasonable to include `jakarta.persistence` as a dependency there in `polaris-service`.